cobalt/common/icu_init/init.cc

```diff
@@ -0,0 +1,115 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fcntl.h>
+#include <pthread.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unicode/utypes.h>
+#include <unistd.h>
+
+#include <string>
+#include <vector>
+
+#include "starboard/common/log.h"
+#include "starboard/configuration_constants.h"
+#include "starboard/file.h"
+#include "starboard/system.h"
+#include "unicode/putil.h"
+#include "unicode/udata.h"
+
+namespace cobalt {
+namespace common {
+namespace icu_init {
+
+namespace {
+
+constexpr std::string_view kIcuDataFileName = "icudtl.dat";
+
+pthread_once_t g_initialization_once = PTHREAD_ONCE_INIT;
+
+void PrintIcuNotLoadedWarning() {
+  SB_LOG(WARNING) << "ICU Database could not be loaded!";
+}
+
+// Initialize the ICU using the file named kIcuDataFileName in the
+// kSbSystemPathContentDirectory folder. Note that this gives the ICU its
+// database, but it does not actually set a default timezone or locale.
+void InitializeIcuDatabase() {
```

Comment:
This function would be more readable if it had helper functions to 
1. Get the icu path
2. Read the icu file
3. Call madvise/set the data (potentially within the function itself)